compiler/noirc_frontend/src/parser/errors.rs

```diff
@@ -92,6 +92,10 @@ impl ParserError {
     pub fn span(&self) -> Span {
         self.span
     }
+
+    pub fn reason(&self) -> Option<ParserErrorReason> {
+        self.reason.clone()
```

Comment:
Can we avoid the clone here and return `self.reason.as_ref()` instead?

Reply:
Fixed